src/uct/ib/base/ib_device.c

```diff
@@ -80,6 +81,43 @@ typedef struct uct_ib_device_subnet {
 UCS_ARRAY_DECLARE_TYPE(uct_ib_device_subnet_array_t, unsigned,
                        uct_ib_device_subnet_t);
 
+typedef struct {
+    uct_ib_device_t *dev;
+    uint8_t          port_num;
+    int              gid_index;
+} uct_ib_device_to_ndev_key_t;
+
+typedef struct {
+    char ndev_name[IFNAMSIZ];
```

Comment:
why need ndev_name in the value?
probably no need for this struct - can use use int as the hash value